spec/knapsack_pro/repository_adapters/git_adapter_spec.rb

```diff
@@ -31,4 +31,56 @@
     it { expect(subject.include?('master')).to be true }
     it { expect(subject.include?(circle_branch)).to be true } if ENV['CIRCLECI']
   end
+
+  describe '#author' do
+    it "returns the masked author" do
+      allow_any_instance_of(KnapsackPro::RepositoryAdapters::GitAdapter).to receive(:git_author).and_return(
+        "John Doe <john.doe@example.com>" + "\n"
+      )
+
+      subject = KnapsackPro::RepositoryAdapters::GitAdapter.new
+
+      expect(subject.author).to eq 'Jo** Do* <jo**.do*@ex*****.co*>'
+    end
+
+    context "when git_author raises an exception" do
+      it "returns the no git author" do
+        allow_any_instance_of(KnapsackPro::RepositoryAdapters::GitAdapter).to receive(:git_author).and_raise(Exception)
+
+        subject = KnapsackPro::RepositoryAdapters::GitAdapter.new
+
+        expect(subject.author).to eq 'no git <no.git@example.com>'
+      end
+    end
+  end
+
+  describe '#authors' do
+    it "returns the masked authors" do
+      allow_any_instance_of(KnapsackPro::RepositoryAdapters::GitAdapter).to receive(:git_authors).and_return([
+        "     5\t3v0k4 <riccardo.odone@gmail.com>\n",
```

Comment:
We could use fake emails here instead of real personal data in a public open source project.